src/librustc_mir/interpret/intrinsics.rs

```diff
@@ -389,6 +389,8 @@ impl<'mir, 'tcx: 'mir, M: Machine<'mir, 'tcx>> InterpCx<'mir, 'tcx, M> {
                 );
                 self.copy_op(self.operand_index(args[0], index)?, dest)?;
             }
+            // FIXME(#73156): Handle source code coverage in const eval
+            sym::count_code_region => (),
```

Comment:
Please always ping @rust-lang/wg-const-eval for new intrinsics in CTFE.
It is not great when I realize by mere accident that we have a new intrinsic here.

Reply:
Also, note that as per this comment, T-lang should have been involved as well:

https://github.com/rust-lang/rust/blob/394e1b40d264aa6928811919c1124fa248e7d802/src/libcore/intrinsics.rs#L8-L9

Reply:
amusingly this intrinsic is *not* marked as const at all. The static checks preventing non-const intrinsics from being invoked happen before mir optimizations. Since instrumentation is a MIR transformation... this isn't caught. Maybe instrumentation should happen before const checking and such?

Reply:
Oh, so this intrinsic is not called by users, but the call is inserted by an instrumentation pass later? Interesting. That reduces my T-lang concern.

> Maybe instrumentation should happen before const checking and such?

Maybe -- I am not sure to what extend it would be useful to check the instrumented code as if it was user-written.

Reply:
If I write

```rust
const fn a() -> u8 {
    42
}

const A: u8 = a();

fn main() {
    println!("{}", A);
}
```

Then currently `a` will contribute towards coverage, yet it is not marked as executed. I think the `count_code_region` intrinsic should also mark code as executed during CTFE.

Reply:
@bjorn3 that is a separate issue though from the other discussion here? Namely, it is https://github.com/rust-lang/rust/issues/73156.

Reply:
Didn't see that issue, thanks.

Reply:
Sorry about that @RalfJung! That was my fault; I didn't realize we should be pinging that group. 

Do you think we still need to get the lang team involved? This intrinsic is used by an unstable `-Z` compiler flag which generates code coverage data using LLVM features. The compiler team already signed off on this approach via an MCP.

Reply:
I doubt they will have any objections, but sending T-lang an "FYI this is what we did here" would still be a good idea I think.

Reply:
Thanks for flagging the issue @RalfJung ...

Note also, the next PR increment is under review, and adds some additional Rust intrinsics that I believe we need for coverage map generation.

See this related note:

https://github.com/rust-lang/rust/pull/73684#discussion_r446559720

Since the MIR analysis is minimal at the moment (injecting at the function level only) I'm not generating calls to these intrinsics yet, but when I do, I suspect I may need to add these new intrinsics to src/librustc_mir/interpret/intrinsics.rs as well, similarly stubbed out.